platform/ios/app/MBXStateManager.m

```diff
@@ -0,0 +1,35 @@
+#import "MBXStateManager.h"
+#import <Mapbox/Mapbox.h>
+
+@implementation MBXStateManager
+
++ (instancetype) sharedManager {
+    static dispatch_once_t once;
+    static MBXStateManager* sharedManager;
+    dispatch_once(&once, ^{
+        sharedManager = [[self alloc] init];
+        sharedManager.currentState = [[MBXState alloc] init];
+    });
+
+    return sharedManager;
+}
+
+- (MBXState*)currentState {
+    return _currentState;
+}
+
+- (void)saveState:(MBXState*)state forKey:(NSObject*)key {
+
+}
+
+-(void)restoreState {
+    // Load from NSUserDefaults
+}
+
+-(void)resetState {
+    self.currentState = nil;
```

Comment:
What do you think about having this delete the stored state, and apply any registered defaults (like the removed `registerDefaults` call below, but moved into this manager)?